src/main/editDelete/preProcess.ts

```diff
@@ -50,6 +31,74 @@ const injectAttributes: MapCallback<PartialWord, Word> = (word, i) => ({
   fileName: 'PLACEHOLDER FILENAME',
 });
 
+const constructWord = (
+  word: string,
+  startTime: number,
+  duration: number,
+  outputStartTime: number,
+  key: string
+) => {
+  const newWord: Word = {
+    word,
+    startTime: roundToMs(startTime),
+    duration: roundToMs(duration),
+    outputStartTime: roundToMs(outputStartTime),
+    deleted: false,
+    key,
+    fileName: 'PLACEHOLDER FILENAME',
+  };
+
+  return newWord;
+};
+
+/**
+ * Adds spaces between words which represent the silence between each word
+ * @param result The output being constructed
+ * @param word The current element of words
+ * @param index The index of word in words
+ * @param words The list of words being reduced
+ * @returns The updated transcript with a silence after word
+ */
+let TOTAL_DURATION = 0;
+const addSpaces = (
+  result: Word[],
+  word: Word,
+  index: number,
+  words: Word[]
+) => {
+  const spaceChar = ' ';
+
+  // is the first word
+  if (index === 0) {
+    result.push(
+      constructWord(spaceChar, 0, words[index].startTime, 0, index.toString())
+    );
+  }
+
+  const isLastWord = index === words.length - 1;
+  let silenceDuration;
+  if (isLastWord) {
```

Comment:
Might be simpler to use an intermediate variable defined with a ternary, that way you can make silenceDuration a const

i.e.
`const endTime = isLastWord ? TOTAL_DURATION : words[index + 1].startTime`
then use endTime in both cases

Reply:
cool trick, addressed here: https://github.com/patrickbrett/mlvet/pull/54/commits/c0d453bab2060cc1c723a21702e63c7782237f36